src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -331,6 +335,8 @@ internal ImmutableArray<IArgumentOperation> DeriveArguments(BoundNode containing
                 return ImmutableArray<IArgumentOperation>.Empty;
             }
 
+            Debug.Assert(optionalParametersMethod is not null);
```

Comment:
>Debug.Assert(optionalParametersMethod is not null); [](start = 12, length = 51)

Would it be better (more robust) to simply check `optionalParametersMethod is null || ...` in the `if` above?

Reply:
It would certainly be less likely to crash, but it's likely that would be masking a bug, which could then result in downstream bugs in analyzers.

---
In reply to: [497823219](https://github.com/dotnet/roslyn/pull/48168#discussion_r497823219) [](ancestors = 497823219)